### PR TITLE
(fix): Fix vault integration to work with rotated keys

### PR DESCRIPTION
**Pull request by rjbrown57**
Signed-off-by: Russell Brown <rjbrown57@gmail.com>

Fixes https://github.com/sigstore/cosign/issues/890

* Use regex instead of trimPrefix to fix signing
* Introduce environment variable VAULT_KEY_PREFIX to allow setting of which key version to verify with. If not specified will default to "vault:v1:"

### CODE REVIEW

File: pkg/signature/kms/hashivault/client.go [hunk 1]
```diff
@@ -48,7 +47,7 @@ var (
 )
 
 const (
-	vaultV1DataPrefix = "vault:v1:"
+	vaultV1DataRegExPrefix = "vault:v[0-9]+:"
```
Comment (bobcallaway) [inline pkg/signature/kms/hashivault/client.go:50]: Nit: use `regexp.MustCompile` as above so we don’t recompile the regex every time we call `vaultDecode`

File: pkg/signature/kms/hashivault/client.go [hunk 2]
```diff
@@ -200,9 +199,14 @@ func (h hashivaultClient) verify(sig, digest []byte, alg crypto.Hash) error {
 	client := h.client.Logical()
 	encodedSig := base64.StdEncoding.EncodeToString(sig)
 
+	vaultDataPrefix := os.Getenv("VAULT_KEY_PREFIX")
+	if vaultDataPrefix == "" {
+		vaultDataPrefix = "vault:v1:"
```
Comment (bobcallaway) [inline pkg/signature/kms/hashivault/client.go:204]: Define this as a `const` above?

**Review by bobcallaway — COMMENTED** [review]
LGTM minus two small nits

File: pkg/signature/kms/hashivault/client.go (hunk 2 above)
Comment (rjbrown57) [reply pkg/signature/kms/hashivault/client.go]: Adjusted as requested. Thanks for the good feedback!

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**